src/models/model.js

```diff
@@ -83,22 +80,87 @@ export default class Model {
 
     // If it's a partial update, get only the fields specified
     if (fields.length > 0) {
-      fields.forEach(field => {
+      fields.forEach((field) => {
         data[field] = obj[field]
       })
     } else {
       data = { ...obj }
     }
 
     // Delete private '_fields' member
-    delete data['_fields'];
-
+    delete data['_fields']
+    // format it the way it is expected
+    Object.keys(data).forEach((k) => {
+      if (
+        data[k] instanceof Model ||
+        (Array.isArray(data[k]) && data[k].length && data[k][0] instanceof Model)
+      ) {
+        if (Array.isArray(data[k])) {
+          data[k] = data[k].map((value) => value.constructor.toAPI(value))
+        } else data[k] = data[k].constructor.toAPI(data[k])
+      }
+    })
     // Remove read only and excluded fields
-    [...this.getReadOnlyFields(), ...excludeFields].forEach(item => { delete data[item] })
-
+    ;[...this.getReadOnlyFields(), ...excludeFields].forEach((item) => {
+      delete data[item]
+    })
     return objectToSnakeCase(data)
   }
 
+  duplicate() {
+    const fields = {}
+    const modelFields = {}
+
+    for (const prop in this) {
+      if (prop !== '_fields' && this._fields[prop]) {
+        if (this._fields[prop] instanceof ModelField && notNullOrUndefined(this[prop])) {
+          if (Array.isArray(this._fields[prop])) {
+            modelFields[prop] = this[prop].map((field) => field.duplicate())
+          } else {
```

Comment:
hmm this is unclear will update it before merge, this should be a field that is of type ModelClass so maybe something like modelField is a bit clearer as the duplicate method only exists on the ModelClass